src/Mvc/Mvc.Core/src/Builder/ControllerEndpointRouteBuilderExtensions.cs

```diff
@@ -152,6 +153,296 @@ public static IEndpointConventionBuilder MapDefaultControllerRoute(this IEndpoin
+        /// requests for non-filenames with a very low priority. The request will be routed to a controller endpoint that
+        /// matches <paramref name="action"/>, and <paramref name="controller"/>.
+        /// </summary>
+        /// <param name="routes">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="action">The action name.</param>
+        /// <param name="controller">The controller name.</param>
+        /// <remarks>
+        /// <para>
+        /// <see cref="MapFallbackToController(IEndpointRouteBuilder, string, string)"/> is intended to handle cases where URL path of
+        /// the request does not contain a filename, and no other endpoint has matched. This is convenient for routing
+        /// requests for dynamic content to a SPA framework, while also allowing requests for non-existent files to
+        /// result in an HTTP 404.
+        /// </para>
+        /// <para>
+        /// <see cref="MapFallbackToController(IEndpointRouteBuilder, string, string)"/> registers an endpoint using the pattern
+        /// <c>{*path:nonfile}</c>. The order of the registered endpoint will be <c>int.MaxValue</c>.
+        /// </para>
+        /// <para>
+        /// <see cref="MapFallbackToController(IEndpointRouteBuilder, string, string)"/> does not re-execute routing, and will
+        /// not generate route values based on routes defined elsewhere. When using this overload, the <c>path</c> route value
+        /// will be available. 
+        /// </para>
+        /// <para>
+        /// <see cref="MapFallbackToController(IEndpointRouteBuilder, string, string)"/> does not attempt to disambiguate between
+        /// multiple actions that match the provided <paramref name="action"/> and <paramref name="controller"/>. If multiple
+        /// actions match these values, the result is implementation defined.
+        /// </para>
+        /// </remarks>
+        public static void MapFallbackToController(
+            this IEndpointRouteBuilder routes, 
+            string action, 
+            string controller)
+        {
+            if (routes == null)
+            {
+                throw new ArgumentNullException(nameof(routes));
+            }
+
+            if (action == null)
+            {
+                throw new ArgumentNullException(nameof(action));
+            }
+
+            if (controller == null)
+            {
+                throw new ArgumentNullException(nameof(controller));
+            }
+
+            EnsureControllerServices(routes);
+
+            // Called for side-effect to make sure that the data source is registered.
+            GetOrCreateDataSource(routes);
+
+            // Maps a fallback endpoint with an empty delegate. This is OK because
+            // we don't expect the delegate to run. 
+            routes.MapFallback(context => Task.CompletedTask).Add(b =>
```

Comment:
nit: Have a shared EmptyRequestDelegate property for `context => Task.CompletedTask` and reuse

Reply:
I haven't done that because there really isn't a good place for that to live that wants to be public. It's static anyway so this is already *free*